src/EditorFeatures/CSharpTest/CompleteStatement/CSharpCompleteStatementCommandHandlerTests.cs

```diff
@@ -0,0 +1,2577 @@
+@"
+class ContinueTest
+{
+    static void Main()
+    {
+        for (int i = 1; i <= 10; i++)
+        {
+            if (i < 9)
+            {
+                continue$$
+            }
+            Console.WriteLine(i);
+        }
+    }
+}";
+
+            VerifyNoSpecialSemicolonHandling(code);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.CompleteStatement)]
+        public void DontComplete_Continue2()
+        {
+            var code =
+@"
+class ContinueTest
+{
+    static void Main()
+    {
+        for (int i = 1; i <= 10; i++)
+        {
+            if (i < 9)
+            {
+                cont$$inue
+            }
+            Console.WriteLine(i);
+        }
+    }
+}";
+
+            VerifyNoSpecialSemicolonHandling(code);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.CompleteStatement)]
+        public void DontComplete_Continue3()
+        {
+            var code =
+@"
+class ContinueTest
+{
+    static void Main()
+    {
+        for (int i = 1; i <= 10; i++)
+        {
+            if (i < 9)
+            {
+                $$continue
+            }
+            Console.WriteLine(i);
+        }
+    }
+}";
+
+            VerifyNoSpecialSemicolonHandling(code);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.CompleteStatement)]
+        public void DontComplete_GoTo()
+        {
+            var code =
+@"
+static void Main()
+{
+    int n = 1;
+    switch (n)
+    {
+        case 1:
+            goto $$case 3;                
+            break;
+        case 3:
+            break
+        default:
+            break;
+    }
+}
+";
+
+            VerifyNoSpecialSemicolonHandling(code);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.CompleteStatement)]
+        public void DontComplete_IfStatement()
+        {
+            var code =
+@"
+class Program
+{
+    void M()
+    {
+        int x = 0;
+        if (x == 0$$)
+        {
+            return;
+        }
+    }
+}
+";
+
+            VerifyNoSpecialSemicolonHandling(code);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.CompleteStatement)]
+        public void DontComplete_Labeled()
+        {
+            var code =
+@"
+class Program
+{
+    static void Main()
+    {
+        if (true)
+            goto labeled;
+        labeled$$: return;
+    }
+}
+";
+
+            VerifyNoSpecialSemicolonHandling(code);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.CompleteStatement)]
+        public void DontComplete_IfStatement2()
+        {
+            var code =
+@"
+class Program
+{
+    void M()
+    {
+        int x = 0;
+        if (x == Math.Min(4,5$$))
+        {
+            return;
+        }
+    }
+}
+";
+
+            VerifyNoSpecialSemicolonHandling(code);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.CompleteStatement)]
+        public void DontComplete_ClassNameOfMethodInvocation1()
```

Comment:
>DontComplete_ClassNameOfMethodInvocation1 [](start = 20, length = 41)

Shouldn't this test have original code snippet with a missing semicolon, i.e. `@"var test = $$ClassC.MethodM(x,y)"`? Otherwise, there is nothing to complete here.

Reply:
Same question for bunch of tests below.

---
In reply to: [242665959](https://github.com/dotnet/roslyn/pull/30743#discussion_r242665959) [](ancestors = 242665959)

Reply:
Yes, I fixed this